react/src/js/react/posts/CreatePost.jsx

```diff
@@ -0,0 +1,124 @@
+/* eslint-disable one-var */
+import React from "react";
+import Reflux from "reflux";
+
+import { FormGroup, FormControl, Form, Button, Row, Col } from "react-bootstrap";
+import PrivacyDropdown from "./PrivacyDropdown";
+
+import { PostsStore, PostsActions } from "./PostsStore";
+import AuthStore from "../auth/AuthStore";
+
+export default class CreatePost extends Reflux.Component {
+    constructor(props) {
+        super(props);
+        this.stores = [PostsStore, AuthStore];
+        this.state = {
+            privacyKey: "PUBLIC"
+        };
+    }
+
+    handleSubmit = (e) => {
+        e.preventDefault();
+        const form = e.currentTarget;
+        const files = form.elements.fileUpload.files;
+
+        // eslint-disable-next-line no-undef
+        const formData = new FormData();
+        Array.from(files).forEach((file) => {
+            formData.append("imageFiles", file);
+        });
+
+        let categories = form.elements.categories.value;
+        if (!categories) {
+            categories = [];
+        }
+
+        const data = {
+            title: form.elements.title.value,
+            content: form.elements.content.value,
+            source: form.elements.source.value,
+            origin: form.elements.origin.value,
+            contentType: form.elements.contentType.value,
+            description: form.elements.description.value,
+            unlisted: form.elements.unlisted.value, // TODO Fix this
+            categories: categories,
+            visibility: this.state.privacyKey,
+            author: this.state.userId
+        };
+        formData.append("postData", JSON.stringify(data));
+        formData.append("query", "createPost");
+        PostsActions.createPost(formData);
+        this.props.handleClose();
```

Comment:
Wouldn't it make more sense to wait for the post to be successfully created before closing the modal? That would let us show a loading icon or an error message as well